### PR TITLE
shibboleth: depend on apr, apr-util for Sierra

### PR DESCRIPTION
**Pull request by FranklinChen**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally prior to submission with `brew install <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [x] Does your submission pass `brew audit --new-formula <formula>` (after doing `brew install <formula>`)?

---

Resolve https://github.com/Homebrew/homebrew-core/issues/5339

### CODE REVIEW

**Comment by ilovezfs** [comment]
This is probably only needed at build time (`=> :build`). Also, it should probably be scoped to the systems that actually need it (e.g., `DevelopmentTools.clang_build_version >= 800).

**Comment by FranklinChen** [comment]
I copied what `passenger` does for build scoping.

File: Formula/shibboleth-sp.rb [hunk 1]
```diff
@@ -11,7 +11,7 @@ class ShibbolethSp < Formula
 
   option "with-apache-22", "Build mod_shib_22.so instead of mod_shib_24.so"
 
-  depends_on :macos => :yosemite
+  depends_on macos: :yosemite
```
Comment (MikeMcQuaid) [inline Formula/shibboleth-sp.rb]: Keep this as it was. Apologies that this has changed recently.

File: Formula/shibboleth-sp.rb [hunk 2]
```diff
@@ -43,7 +48,19 @@ def install
...
-  plist_options :startup => true, :manual => "shibd"
+  def caveats
+    mod = (build.with? "apache-22")? "mod_shib_22.so" : "mod_shib_24.so"
+    <<-EOS.undent
+      You must manually edit httpd.conf to include
+      LoadModule mod_shib #{lib}/shibboleth/#{mod}
+      You must also manually configure
+        #{etc}/shibboleth/shibboleth2.xml
+      as per your own requirements. For more information please see
+        https://wiki.shibboleth.net/confluence/display/EDS10/3.1+Configuring+the+Service+Provider
```
Comment (MikeMcQuaid) [inline Formula/shibboleth-sp.rb]: Let's leave these as they contain Homebrew-specific paths but the instructions aren't Homebrew-only.

File: Formula/shibboleth-sp.rb [hunk 3]
```diff
@@ -43,7 +48,19 @@ def install
     system "make", "install"
   end
 
-  plist_options :startup => true, :manual => "shibd"
```
Comment (MikeMcQuaid) [inline Formula/shibboleth-sp.rb]: Leave this as-is.

**Review by MikeMcQuaid — CHANGES_REQUESTED** [review]

**Comment by DomT4** [comment]
> This is probably only needed at build time (=> :build). Also, it should probably be scoped to the systems that actually need it (e.g., DevelopmentTools.clang_build_version >= 800).

It won't be build-time only. That's only really the case when using the system version of `apr`, and even then it's a hack around our dependency system more than anything else. Most stuff does dynamically link to the Apr library. Someone might just want to change the requirement in `brew` now this is expanding beyond `subversion` as a problem.

`DevelopmentTools.clang_build_version >= 800` isn't strictly true IIRC; they'll still be available via the 7.x CLT on El Cap, so even if you've Xcode 8 on 10.11 as long as the CLT is installed you'll be fine. Off the top of my head though, so feel free to check.

**Comment by ilovezfs** [comment]
`DevelopmentTools.clang_build_version >= 800` is what's reported if both Xcode 8 and CLT 7 are installed if that's what you're asking.

**Comment by DomT4** [comment]
Anything `:` there is fine as `:build`, kind of, that's a hack around our dep system IIRC. Anything `""` should be checked, but I'd be very surprised if they were build unless those formulae only produce static libraries/executables.

**Comment by ilovezfs** [comment]
@DomT4 LOL https://github.com/Homebrew/homebrew-core/issues/5599

**Comment by DomT4** [comment]
Yeah. Probably worth giving up doing this individually and just enforce it for Sierra via the requirement.

**Comment by MikeMcQuaid** [comment]
Blocked on https://github.com/Homebrew/brew/pull/1227.

File: Formula/shibboleth-sp.rb [hunk 4]
```diff
@@ -21,6 +21,11 @@ class ShibbolethSp < Formula
   depends_on "boost"
   depends_on "unixodbc"
 
+  if MacOS.version >= :sierra
+    depends_on "apr-util" => :build
+    depends_on "apr" => :build
```
Comment (MikeMcQuaid) [inline Formula/shibboleth-sp.rb]: Make these unconditional.

Comment (MikeMcQuaid) [inline Formula/shibboleth-sp.rb]: Also, are you sure they aren't linked against?

**Review by MikeMcQuaid — CHANGES_REQUESTED** [review]

**Comment by MikeMcQuaid** [comment]
@FranklinChen Any news on the comments above? Thanks!

**Comment by FranklinChen** [comment]
I'd been busy, will address this.

**Comment by FranklinChen** [comment]
I updated according to comments here.

File: Formula/shibboleth-sp.rb (hunk 2 above)
Comment (MikeMcQuaid) [inline Formula/shibboleth-sp.rb]: Actually, they were there before so it's fine.

**Review by MikeMcQuaid — APPROVED** [review]

**Comment by MikeMcQuaid** [comment]
Thanks again for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

**Comment by FranklinChen** [comment]
Is this going to be merged?

**Comment by MikeMcQuaid** [comment]
It was.

**Pull request closed without merging**